devtools/maven/src/main/java/io/quarkus/maven/GenerateConfigMojo.java

```diff
@@ -132,6 +132,8 @@ public void execute() throws MojoExecutionException {
             try (CuratedApplication curatedApplication = QuarkusBootstrap
                     .builder()
                     .setAppArtifact(appArtifact)
+                    .setApplicationRoot(classesDir)
+                    .setProjectRoot(project.getBasedir().toPath())
```

Comment:
Yes, tests create their own tmp dir which is the app root, but uses metadata from the project root.

Reply:
I meant in this specific case. I don't think all these have to required. It does not look simpler to me.

Reply:
I want it to be explicit exactly which paths are used where. At the moment if one is null it just picks a different one. We only have a few different places that use this launcher code, IMHO it is better to be explicit rather than rely on implicit behaviour.